odk/src/main/java/org/motechproject/odk/domain/builder/FormInstanceBuilder.java

```diff
@@ -0,0 +1,160 @@
+        formInstance.setFormValues(formValues);
+        return formInstance;
+    }
+
+    private FormValue buildFormElementValueByType(FormElement formElement, Object value) {
+
+        switch (formElement.getType()) {
+
+            case FieldTypeConstants.REPEAT_GROUP:
+                return buildGroup(formElement, value);
+
+            case FieldTypeConstants.INT:
+                return new FormValueInteger(formElement.getName(), formElement.getLabel(), formElement.getType(), (Integer) value);
+
+            case FieldTypeConstants.DECIMAL:
+                return new FormValueDouble(formElement.getName(), formElement.getLabel(), formElement.getType(), (Double) value);
+
+            case FieldTypeConstants.DATE_TIME:
+                return new FormValueDateTime(formElement.getName(), formElement.getLabel(), formElement.getType(), (DateTime) value);
+
+            case FieldTypeConstants.DATE:
+                return new FormValueDateTime(formElement.getName(), formElement.getLabel(), formElement.getType(), (DateTime) value);
+
+            case FieldTypeConstants.TIME:
+                return new FormValueDateTime(formElement.getName(), formElement.getLabel(), formElement.getType(), (DateTime) value);
+
+            case FieldTypeConstants.SELECT:
+                return new FormValueStringList(formElement.getName(), formElement.getLabel(), formElement.getType(), (List<String>) value);
+
+            case FieldTypeConstants.STRING_ARRAY:
+                return new FormValueStringList(formElement.getName(), formElement.getLabel(), formElement.getType(), (List<String>) value);
+
+            default:
+                return new FormValueString(formElement.getName(), formElement.getLabel(), formElement.getType(), (String) value);
+        }
+    }
+
+    private FormValueGroup buildGroup(FormElement formElement, Object value) {
+        List<Map<String, Object>> data;
+
+        if (value instanceof String) {
+            data = jsonToMap((String) value);
+
+        } else {
+            data = (List<Map<String, Object>>) value;
+        }
+
+        List<FormValue> children = new ArrayList<>();
+        for (Map<String, Object> map : data) {
+
+            for (Map.Entry pair : map.entrySet()) {
+                FormElement element = findFormElementByName(pair.getKey().toString());
+                Object pairValue = pair.getValue();
+                children.add(buildFormElementValueByType(element, pairValue));
+            }
+
+        }
+
+        return new FormValueGroup(formElement.getName(), formElement.getLabel(), formElement.getType(), children);
+    }
+
+    private List<Map<String, Object>> jsonToMap(String json) {
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            return mapper.readValue((String) json, new TypeReference<List<Map<String, Object>>>() {
+            });
+
+        } catch (Exception e) {
```

Comment:
is this a good idea?  All this will lead to is a null pointer exception later on. Why not throw a descriptive exception here?